OpenRA.Mods.Common/Traits/World/EditorSelectionLayer.cs

```diff
@@ -0,0 +1,96 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.IO;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Required for the map editor to work. Attach this to the world actor.")]
+	public class EditorSelectionLayerInfo : ITraitInfo
+	{
+		[Desc("Palette to use for rendering the placement sprite.")]
+		public readonly string Palette = "terrain";
+
+		[Desc("Sequence image where the selection overlay types are defined.")]
+		public readonly string Image = "editor-overlay";
+
+		public virtual object Create(ActorInitializer init) { return new EditorSelectionLayer(init.Self, this); }
+	}
+
+	public class EditorSelectionLayer : IWorldLoaded, IPostRender
+	{
+		readonly EditorSelectionLayerInfo info;
+		readonly Map map;
+		readonly Sprite copySprite;
+		readonly Sprite pasteSprite;
+		PaletteReference palette;
+
+		public CellRegion CopyRegion { get; private set; }
+		public CellRegion PasteRegion { get; private set; }
+
+		public EditorSelectionLayer(Actor self, EditorSelectionLayerInfo info)
+		{
+			if (self.World.Type != WorldType.Editor)
+				return;
+
+			this.info = info;
+			map = self.World.Map;
+			copySprite = map.SequenceProvider.GetSequence(info.Image, "copy").GetSprite(0);
+			pasteSprite = map.SequenceProvider.GetSequence(info.Image, "paste").GetSprite(0);
```

Comment:
Please expose them to lint testing with `[SequenceReference("Image")]` in the `EditorSelectionLayerInfo` above.